framework/Models/Code.py

```diff
@@ -270,11 +270,15 @@ def initialize(self,runInfoDict,inputFiles,initDict=None):
     self.workingDir               = os.path.join(runInfoDict['WorkingDir'],runInfoDict['stepName']) #generate current working dir
     runInfoDict['TempWorkingDir'] = self.workingDir
     for inputFile in inputFiles:
-      shutil.copy(inputFile.getAbsFile(),self.workingDir)
+      subSubDirectory = os.path.join(self.workingDir,inputFile.subDirectory)
+      if inputFile.subDirectory.strip() != "" and not os.path.exists(subSubDirectory):
+        os.mkdir(subSubDirectory)
+      shutil.copy(inputFile.getAbsFile(),subSubDirectory)
     self.oriInputFiles = []
     for i in range(len(inputFiles)):
```

Comment:
These loops (line 272-276 & 275-281) look very similar, is there a reason that they cannot be merged into one?

```python
    self.oriInputFiles = []
    for inputFile in inputFiles:
        subSubDirectory = os.path.join(self.workingDir,inputFile.subDirectory)
        if inputFile.subDirectory.strip() != "" and not os.path.exists(subSubDirectory):
            os.mkdir(subSubDirectory)
        shutil.copy(inputFile.getAbsFile(),subSubDirectory)
        self.oriInputFiles.append(copy.deepcopy(inputFile))
        self.oriInputFiles[-1].setPath(subSubDirectory)

```